### DISCUSSION TITLE
Separating PathSearchSolution from Forward and from Backward solvers.

### DISCUSSION THREAD

**Discussion by pevnak** [JuliaPlanners/SymbolicPlanners.jl#11] [Ideas]
Hi,

I wonder if it is possible at the moment to identify, from which solver the `PathSearchSolution` comes from. I guess the only way is to test, if first (`start`) and last(`goal`) state in trajectory have parents, but it is a bit hacky. Since the `search_tree` is effectively in different order, would not make sense to add flag to `PathSearchSolution` (or flag to the type of PathSearchSolution) or use separate types?

Tomas

**Comment by ztangent** [comment]
That's certainly one way of testing it! In the future I think it could be made clear from creating a `PartialState` type that the backward planner will use and store. But what is the use case for making this distinction?

**Comment by pevnak** [comment]
I am playing a bit with bootstrapping, where I am optimizing heuristics (realized by NN) from problems it has solved and for construction, I need to know the direction of the search tree, such that I know, where is the goal state.

I think for now, I will use the heuristic I have figured out when I was asking my questions above.
